src/types.rs

```diff
@@ -921,3 +921,103 @@ impl Debug for Log {
             .finish()
     }
 }
+
+/// An Ethereum Transaction Receipt
+#[derive(Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionReceipt {
+    /// The type of the transaction. These have evolved over several EIPs;
+    /// See `TransactionReceiptKind` for summaries and references.
+    #[serde(flatten)]
+    pub kind: TransactionReceiptKind,
+    /// The hash of the transaction that emitted this log.
+    pub transaction_hash: Digest,
+    /// The index of the transaction that emitted this log within the block.
+    pub transaction_index: U256,
+    /// The hash of the block containing the transaction.
+    pub block_hash: Digest,
+    /// The height of the block containing the transaction.
+    pub block_number: U256,
+    /// Address of transaction sender.
+    pub from: Address,
+    /// Transaction receipient ([`None`] for contract creation).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to: Option<Address>,
+    /// The price paid post-execution by the transaction (i.e. base fee + priority fee).
+    /// Both fields in 1559-style transactions are *maximums* (max fee + max priority fee), the
+    /// amount that's actually paid by users can only be determined post-execution.
+    pub effective_gas_price: U256,
+    /// The sum of gas used by this transaction and all preceding transactions in the same block.
+    pub cumulative_gas_used: U256,
+    /// The amount of gas used for this specific transaction alone.
+    pub gas_used: U256,
+    /// Contract address created, or `None` if not a deployment.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contract_address: Option<Address>,
+    /// Logs emitted by the transaction.
+    pub logs: Vec<Log>,
+    /// The log bloom filter.
+    pub logs_bloom: Bloom,
+    /// The transaction status, indicating whether it succeeded or reverted.
+    pub status: TransactionReceiptStatus,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub enum TransactionReceiptStatus {
+    /// Status of a failed transaction.
+    #[serde(rename = "0x0")]
+    Failure,
+    /// Status of a successful transaction.
+    #[serde(rename = "0x1")]
+    Success,
+}
+
```

Comment:
```suggestion

/// The type of a `TransactionReceipt`.
```

Reply:
I just realized that we now have TransactionType enum for the Transaction Struct and TransactionReceipt type for TransactionReceipt struct. It seems that these could be unified (although I think the existing one might be outdated).

Reply:
Yes, maybe they can. It depends if you can specify the same blob fields with the transaction kind or not.

I would keep it it outside of this PR as it is a bit unrelated.